### ISSUE TITLE
Predicting an 8 hour frequency time series using multiple past covariate time series with 5min frequencies

### ISSUE THREAD

**Issue by alitirmizi23** [unit8co/darts#487]
Hi all. Great work. Just wanted to know if there is currently a way to do forecasting for a time series with values arriving every 8 hours using multiple past covariate time series (all of which have 5min frequencies). So let's say I want n=3 steps ahead forecasts, they would actually be for 8hrs, 16hrs and 24hrs ahead in time

**Comment by dennisbader** [comment]
Hi @alitirmizi23 and thanks for the question.

Yes, you can use multiple past covariate time series in your forecasting example. 
I would suggest to try resampling your past covariate time series to the same frequency as the target series (freq='8H') and aggregate over the intervals.

As far as I know, the aggregation part cannot yet be handled by Darts (@hrzn , is that correct?). This needs to be done before converting your data into a TimeSeries.

Let's say your covariates are stored in the columns of a pandas.DataFrame `df_covariates` with an index of type `pandas.DatetimeIndex`. You can resample and aggregate over an '8H' frequency and then convert them to a multivariate TimeSeries like this:
`ts_covariates = TimeSeries.from_dataframe(df_covariates.resample('8H').sum())`

Now you can follow our documentation on [Multiple Time Series, Pre-trained Models and Covariates](https://unit8co.github.io/darts/examples/02-multi-time-series-and-covariates.html#Multiple-Time-Series,-Pre-trained-Models-and-Covariates).

Having all of the TimeSeries with an 8-hour frequency makes the prediction also very easy. After training your model, you can use `model.predict(n=3, ...)` to make a forecast for the next 8, 16, and 24 hour timestamps.

**Comment by alitirmizi23** [comment]
Hi @dennisbader. Thanks for the response. This might actually work; I'll try and let you know. But it kind of does away with the granular information I have - instead of utilizing every 5min reading, I'm aggregating over 8H which is loss of information (in some ways), don't you think? 
I know I could create a 3D ts of my own for a recurrent neural network to train (n_samples, window_size, n_covariates) vs every 8H y values.. but this is also another use case and would be great to have in darts.

**Comment by dennisbader** [comment]
To my knowledge, in Darts, the target series and covariate series currently have to share the same frequency so that the elements of your covariates series can be mapped to the corresponding `y` values with the same timestamp (I'm rather new on this project, but @hrzn, @pennfranc, @TheMP will know more about that :) ) 

About loosing information, it also depends a bit on the underlying problem and what your `y` values describe. 
- For example, if your `y` values are an aggregate of "what happened" over the last 8 hours, then I don't think you loose information as the individual covariates are aggregated separately. 
- If your `y` values just describe a state at the exact time (let's say for the exact '5min' interval at which you receive the information), then extracting only the corresponding '5min' intervals from your covariates could also work.
- Another way would be to upsample your `y` series to '5min' frequency with i.e. repeated or interpolated values. Then you would have to predict n=(24h * 60min/5min)=288 steps and extract the values at the correct 8, 16 and 24 hour indexes.

I don't think that Darts supports the specific RNN use case you describe at the moment. But this will definitely be a good feature to add, thanks!

**Comment by hrzn** [comment]
Here are my two cents here. Indeed you have mainly two solutions, as outlined by @dennisbader .
* Resample everything to `8H` frequency. This is indeed a loss of information, which can potentially hurt (or help!) subsequent models.
* Keep the original 5-minutes granularity and forecast for a longer horizon `n=3*12*8`. This will allow models to have access to higher frequency data, which may or may not be good. Probably the best is to try it.
Currently Darts requires covariates and targets to have the same frequency.

Hope this helps.
